tests/test_superruns.py

```diff
@@ -0,0 +1,247 @@
+import os
+import unittest
+import shutil
+import strax
+import numpy as np
+import tempfile
+from strax.testutils import Records, PeaksWoByRunDefault
+import datetime
+import pytz
+import json
+from bson import json_util
+import re
+
+
+class TestSuperRuns(unittest.TestCase):
+
+    def setUp(self, superrun_name='_superrun_test'):
+        self.offset_between_subruns = 10
+        self.superrun_name = superrun_name
+        self.tempdir = tempfile.mkdtemp()
+        self.context = strax.Context(storage=[strax.DataDirectory(self.tempdir,
+                                                                  provide_run_metadata=True,
+                                                                  readonly=False,
+                                                                  deep_scan=True)],
+                                     register=[Records, RecordsExtension],
+                                     use_per_run_defaults=False)
+        self.context.set_context_config({'write_superruns': True})
+        
+        logger = self.context.log
+        logger.addFilter(lambda s: not re.match(".*Could not estimate run start and end time.*",
+                                                s.getMessage()))
+
+        self._create_subruns()
+        self.context.define_run(self.superrun_name, data=self.subrun_ids)  # Define superrun
+
+    def test_run_meta_data(self):
+        """
+        Check if superrun has the correct run start/end and livetime.
+        """
+        superrun_meta = self.context.run_metadata(self.superrun_name)
+        subrun_meta = [self.context.run_metadata(r) for r in self.subrun_ids]
+
+        assert superrun_meta['start'] == subrun_meta[0]['start']
+        assert superrun_meta['end'] == subrun_meta[-1]['end']
+        livetime = 0
+        for meta in subrun_meta:
+            time_delta = meta['end'] - meta['start']
+            livetime += time_delta.total_seconds()*10**9
+        assert superrun_meta['livetime'] == livetime
+
+    def test_load_superruns(self):
+        """
+        Load superruns from already existing subruns. Does not write
+        "new" data.
+        """
+        self.context.set_context_config({'write_superruns': False})
+        subrun_data = self.context.get_array(self.subrun_ids,
+                                             'records',
+                                             progress_bar=False)
+        superrun_data = self.context.get_array(self.superrun_name, 'records')
+        assert np.all(subrun_data['time'] == superrun_data['time'])
+
+        # Deactivate the lineage check for subruns:
+        superrun_data = self.context.get_array(self.superrun_name,
+                                               'records',
+                                               _check_lineage_per_run_id=False
+                                               )
```

Comment:
This appears to be a function predating #483 right? At least I think `_check_lineage_per_run_id` shouldn't do anything right?

Reply:
There are a few more `_check_lineage_per_run_id` still in this test